### PR TITLE
[Private DC] Various feature update

### PR DESCRIPTION
**Pull request by shifucun**
- Implement /stat/set/within-place API. Also fixed an old bug, that the within-place cache is not used when "date" is provided.
- Fix a bug to check the memory database existence.
- Updated TestOption to be a struct.
- Implement /stat-var/path API.

### CODE REVIEW

**Comment by shifucun** [comment]
@pradh Can you take a look before Monday, thanks!

File: internal/server/stat_point.go [hunk 1]
```diff
@@ -155,80 +155,121 @@ func (s *Server) GetStatSetWithinPlace(
...
+						Metadata: &pb.StatMetadata{ImportName: cohort.ImportName},
 					}
+					result.Data[statVar].Metadata[cohort.ImportName] = metaData
 				}
 			}
 		}
-		if gotResult {
-			return result, nil
+	}
+	// Check if need to read from memory database.
```
Comment (pradh) [inline internal/server/stat_point.go:219]: nit: this was already a long function doing two things, and we're adding a third, might be good to modularize it into functions and comment like 1. read cache x, 2. read memdb, 3. read cache y, ...

File: internal/store/memdb.go [hunk 2]
```diff
@@ -53,6 +63,56 @@ func (memDb *MemDb) ReadSeries(statVar, place string) []*pb.Series {
 	return []*pb.Series{}
 }
 
+// ReadPointValue reads one observation point.
+func (memDb *MemDb) ReadPointValue(statVar, place, date string) (
```
Comment (pradh) [inline internal/store/memdb.go:67]: nit: document function contract based on "date" value?

**Review by pradh — APPROVED** [review]

File: internal/server/stat_point.go (hunk 1 above)
Comment (shifucun) [reply internal/server/stat_point.go:219]: Okay, I will do this in a follow up PR.

**Comment by shifucun** [comment]
thanks for review!

**Pull request merged**